backend/ml/inputs.py

```diff
@@ -0,0 +1,142 @@
+import pandas as pd
+from django.db.models import Case, F, When
+
+from core.models import User
+from tournesol.models import ComparisonCriteriaScore
+from tournesol.models.ratings import ContributorRating
+
+
+class MlInput:
+    def get_comparisons(self, trusted_only=False, criteria=None) -> pd.DataFrame:
+        """Fetch data about comparisons submitted by users
+
+        Returns:
+        - comparisons_df: DataFrame with columns
+            * `user_id`: int
+            * `entity_a`: int or str
+            * `entity_b`: int or str
+            * `criteria`: str
+            * `score`: float
+            * `weight`: float
+        """
+        raise NotImplementedError
```

Comment:
Why NotImplementedError ?

Reply:
Yes. I will define `MlInput` as abstract class to make it more explicit.